watcher/src/watcher.rs

```diff
@@ -360,13 +389,19 @@ impl WatcherSyncThread {
 
             // Maybe sync, maybe wait and check again.
             if is_behind {
-                let max_blocks = std::cmp::min(
-                    ledger_num_blocks - 1,
-                    lowest_next_block_to_sync + MAX_BLOCKS_PER_SYNC_ITERATION as u64,
-                );
-                watcher
-                    .sync_blocks(lowest_next_block_to_sync, Some(max_blocks), true)
+                let sync_result = watcher
+                    .sync_blocks(
+                        lowest_next_block_to_sync,
+                        Some(ledger_num_blocks - 1),
+                        Some(MAX_BLOCKS_PER_SYNC_ITERATION),
+                        true,
+                    )
                     .expect("Could not sync blocks");
+
+                // Pause for a second if a node is having trouble
+                if let SyncResult::BlockSyncError = sync_result {
+                    std::thread::sleep(poll_interval);
```

Comment:
Consider logging a trace like the branch below. Maybe even factor it out into a separate step, e.g.

```rust
let mut sleep = !stop_requested.load(Ordering::SeqCst);

if is_behind {
  let sync_result = ...;
  if let SyncResult::BlockSyncError = sync_result {
    sleep = true;
  }
}

if sleep {
  log::trace!(...)
  sleep(config.poll_interval);
}

Reply:
A trace could go here, but that particular atomic bool is meant to signal a stop flag. Block sync errors shouldn't stop watcher, so that would be the wrong flag to set.